velox/functions/sparksql/fuzzer/tests/SparkQueryRunnerTest.cpp

```diff
@@ -0,0 +1,163 @@
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class SparkQueryRunnerTest : public ::testing::Test,
+                             public velox::test::VectorTestBase {
+ protected:
+  static void SetUpTestCase() {
+    memory::MemoryManager::testingSetInstance({});
+  }
+
+  void SetUp() override {
+    velox::functions::sparksql::registerFunctions("");
+    velox::functions::aggregate::sparksql::registerAggregateFunctions("");
+    velox::parse::registerTypeResolver();
+  }
+};
+
+// This test requires a Spark Coordinator running at localhost, so disable it
+// by default.
+TEST_F(SparkQueryRunnerTest, DISABLED_basic) {
+  auto queryRunner =
+      std::make_unique<fuzzer::SparkQueryRunner>("localhost:15002");
+
+  auto input = makeRowVector({
+      makeConstant<int64_t>(1, 25),
+  });
+  auto outputType = ROW({"a"}, {BIGINT()});
+  auto results =
+      queryRunner->execute("SELECT count(*) FROM tmp", {input}, outputType);
+  auto expected = makeRowVector({
+      makeConstant<int64_t>(25, 1),
+  });
+  exec::test::assertEqualResults(results, outputType, {expected});
+
+  input = makeRowVector({
+      makeFlatVector<int64_t>({0, 1, 2, 3, 4, 0, 1, 2, 3, 4, 0, 1, 2,
+                               3, 4, 0, 1, 2, 3, 4, 0, 1, 2, 3, 4}),
+      makeFlatVector<int64_t>({1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                               1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}),
+  });
+  outputType = ROW({"a", "b"}, {BIGINT(), BIGINT()});
+  results = queryRunner->execute(
+      "SELECT c0, count(*) FROM tmp GROUP BY 1", {input}, outputType);
+  expected = makeRowVector({
+      makeFlatVector<int64_t>({0, 1, 2, 3, 4}),
+      makeFlatVector<int64_t>({5, 5, 5, 5, 5}),
+  });
+  exec::test::assertEqualResults(results, outputType, {expected});
+}
+
+// This test requires a Spark Coordinator running at localhost, so disable it
+// by default.
+TEST_F(SparkQueryRunnerTest, DISABLED_fuzzer) {
+  auto data = makeRowVector({
+      makeFlatVector<int64_t>({1, 2, 3, 4, 5}),
+      makeNullableFlatVector<int64_t>({std::nullopt, 1, 2, std::nullopt, 4, 5}),
+  });
+
+  auto plan = exec::test::PlanBuilder()
+                  .values({data})
+                  .singleAggregation({}, {"sum(c0)", "collect_list(c1)"})
+                  .project({"a0", "array_sort(a1)"})
+                  .planNode();
+
+  auto queryRunner =
+      std::make_unique<fuzzer::SparkQueryRunner>("localhost:15002");
+  auto sql = queryRunner->toSql(plan);
+  ASSERT_TRUE(sql.has_value());
+
+  auto prestoResults = queryRunner->execute(
+      sql.value(), {data}, ROW({"a", "b"}, {BIGINT(), ARRAY(BIGINT())}));
+
+  auto veloxResults = exec::test::AssertQueryBuilder(plan).copyResults(pool());
+  exec::test::assertEqualResults(
+      prestoResults, plan->outputType(), {veloxResults});
+}
+
+TEST_F(SparkQueryRunnerTest, distinctAggregation) {
```

Comment:
perhaps, fold this test into toSql test or update the name to clarify that this test only checks to-sql conversion and doesn't execute the plan.